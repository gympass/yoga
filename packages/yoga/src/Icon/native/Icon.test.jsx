import React from 'react';
import { render } from '@testing-library/react-native';

import { ThemeProvider, Icon } from '../..';

const Circle = props => (
  <svg {...props}>
    <circle cx="16" cy="16" r="16" />
  </svg>
);

describe('Snapshots', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Icon as={Circle} width="small" height="small" fill="stamina" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with size prop', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Icon as={Circle} size="small" fill="stamina" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with system props', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Icon
          as={Circle}
          size="small"
          fill="stamina"
          margin="medium"
          elevation="medium"
        />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
