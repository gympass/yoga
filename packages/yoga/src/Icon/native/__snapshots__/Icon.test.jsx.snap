// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should match snapshot 1`] = `
<svg
  fill="#231B22"
  height={16}
  style={
    Array [
      Object {
        "height": 16,
        "width": 16,
      },
    ]
  }
  width={16}
>
  <circle
    cx="16"
    cy="16"
    r="16"
  />
</svg>
`;

exports[`Snapshots should match snapshot with size prop 1`] = `
<svg
  fill="#231B22"
  height={16}
  style={
    Array [
      Object {
        "height": 16,
        "width": 16,
      },
    ]
  }
  width={16}
>
  <circle
    cx="16"
    cy="16"
    r="16"
  />
</svg>
`;

exports[`Snapshots should match snapshot with system props 1`] = `
<svg
  elevation="medium"
  fill="#231B22"
  height={16}
  margin="medium"
  style={
    Array [
      Object {
        "height": 16,
        "marginBottom": 20,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 20,
        "shadowColor": "rgba(152, 152, 166, 0.2)",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 5,
        "width": 16,
      },
    ]
  }
  width={16}
>
  <circle
    cx="16"
    cy="16"
    r="16"
  />
</svg>
`;
